src/js/strategies/CitrixVDIStrategy.js

```diff
@@ -0,0 +1,131 @@
+import CCPInitiationStrategyInterface from "./CCPInitiationStrategyInterface";
+import {hitch, isCitrixWebRTCSupported} from "../utils";
+import {FailedState} from "../rtc_session";
+import {RTC_ERRORS} from "../rtc_const";
+import "@citrix/ucsdk/CitrixWebRTC";
+
+export default class CitrixVDIStrategy extends CCPInitiationStrategyInterface {
+    constructor() {
+        super();
+        if (!isCitrixWebRTCSupported()) {
+            throw new Error('Citrix WebRTC redirection feature is NOT supported!');
+        }
+        window.getCitrixWebrtcRedir = function() {
+            const registryValue = Promise.resolve(1);
+            return new Promise(function(resolve, reject) {
+                //retrieve registry value internally
+                registryValue.then((v) => {
+                    resolve(v); }).catch(() => {
+                    reject(); })
+            })
+        }
+        window.CitrixWebRTC.initLog(global.connect.getLog());
+        console.log("CitrixVDIStrategy initialized");
+    }
+
+    // the following functions are rtc_peer_connection_factory related functions
+    // check if the browser supports early media connection
+    _isBrowserSupported(){
```

Comment:
can we rename this to _isEarlyMediaConnectionSupported()?

Reply:
Renamed